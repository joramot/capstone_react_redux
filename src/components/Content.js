import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import X from '../images/x-mark.png';
import '../App.css';

const Content = () => {
  const { data, id } = useSelector((store) => store.home);

  return (
    <div>
      <div className="cross-container">
        <Link to="/" className="cross-icon">
          <img src={X} alt="cross" />
        </Link>
      </div>
      <div className="details-container">
        <h1>{data[id].symbol}</h1>
        <h2>{data[id].name}</h2>
        <h3>
          Price:
          {' '}
          {data[id].price}
        </h3>
        <h3>
          Day Low:
          {' '}
          {data[id].dayLow}
        </h3>
        <h3>
          Day High:
          {' '}
          {data[id].dayHigh}
        </h3>
      </div>
    </div>
  );
};

export default Content;
