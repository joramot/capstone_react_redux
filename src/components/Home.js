import { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchData, getId } from '../redux/home/homeSlice';

const Home = () => {
  console.log(fetchData.data);
  const {
    isPending, error, data,
  } = useSelector((state) => state.home);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchData());
  }, [dispatch]);

  const handleGetId = (id) => {
    dispatch(getId(id));
  };

  return (
    <>This the HOME</>
  );
};

export default Home;
