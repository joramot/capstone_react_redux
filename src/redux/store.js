import { configureStore } from "@reduxjs/toolkit";
import homeReducers from '../redux/';

const store = configureStore(
  reducer: {
    home: homeReducers,
  }
);

export default store;