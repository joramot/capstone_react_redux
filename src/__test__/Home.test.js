import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import Home from '../components/Home';
import Content from '../components/Content';

describe('Renders without crashing', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home should render correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/content" element={<Content />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
