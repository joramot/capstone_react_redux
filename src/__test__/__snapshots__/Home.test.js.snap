// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Home should render correctly 1`] = `
<div>
  <h1
    className="header"
  >
    Stock Prices NASDAQ
    <svg
      className="header-icon"
      fill="none"
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.25 6.375c0 2.278-3.694 4.125-8.25 4.125S3.75 8.653 3.75 6.375m16.5 0c0-2.278-3.694-4.125-8.25-4.125S3.75 4.097 3.75 6.375m16.5 0v11.25c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125V6.375m16.5 0v3.75m-16.5-3.75v3.75m16.5 0v3.75C20.25 16.153 16.556 18 12 18s-8.25-1.847-8.25-4.125v-3.75m16.5 0c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </h1>
  <div
    className="search-bar"
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <svg
      className="search"
      fill="none"
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <div
    className="loading"
  >
    <img
      alt="logo"
      className="App-logo"
      src="Arrow-break.png"
    />
    <h2>
       Loading ...
    </h2>
  </div>
  <div
    className="data"
  />
</div>
`;

exports[`Renders without crashing should render correctly 1`] = `
<div>
  <h1
    className="header"
  >
    Stock Prices NASDAQ
    <svg
      className="header-icon"
      fill="none"
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.25 6.375c0 2.278-3.694 4.125-8.25 4.125S3.75 8.653 3.75 6.375m16.5 0c0-2.278-3.694-4.125-8.25-4.125S3.75 4.097 3.75 6.375m16.5 0v11.25c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125V6.375m16.5 0v3.75m-16.5-3.75v3.75m16.5 0v3.75C20.25 16.153 16.556 18 12 18s-8.25-1.847-8.25-4.125v-3.75m16.5 0c0 2.278-3.694 4.125-8.25 4.125s-8.25-1.847-8.25-4.125"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </h1>
  <div
    className="search-bar"
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <svg
      className="search"
      fill="none"
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </div>
  <div
    className="data"
  />
</div>
`;
